### PR TITLE
feat(express,next,node): support fetchUserInfo

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add config "fetchUserInfo" to `express` and `next`, enable user info endpoint in Node.js SDK family.

## Testing

Local tested with express-sample and next-sample.

image

### CODE REVIEW

File: packages/node/src/index.ts [hunk 1]
```diff
@@ -70,6 +70,7 @@ export default class LogtoClient extends BaseClient {
       return {
         isAuthenticated,
         claims,
+        userInfo: await this._fetchUserInfo(fetchUserInfo),
```
Comment (charIeszhao) [inline packages/node/src/index.ts:73]: How about checking and calling `fetchUserInfo` in prior? e.g.
```ts
const userInfo = conditional(fetchUserInfo && (await this.fetchUserInfo()));
```
This way, you don't have to declare a private `_fetchUserInfo` method

File: packages/express-sample/src/app.ts [hunk 2]
```diff
@@ -37,11 +36,23 @@ app.get('/', (request, response) => {
...
 });
 
-app.get('/user', (request, response) => {
+app.get('/user', withLogto(config), (request, response) => {
   response.json(request.user);
 });
 
-app.get('/protected', requireAuth, (request, response) => {
+app.get(
+  '/user-fetched',
```
Comment (charIeszhao) [inline packages/express-sample/src/app.ts:44]: Nit: Maybe `full-user` or `remote-user`?

Comment (wangsijie) [reply packages/express-sample/src/app.ts]: updated

File: packages/node/src/index.ts (hunk 1 above)
Comment (wangsijie) [reply packages/node/src/index.ts]: updated

File: packages/next-sample/pages/index.tsx [hunk 3]
```diff
@@ -5,16 +5,17 @@ import useSWR from 'swr';
 
 const Home = () => {
   const { data } = useSWR<LogtoContext>('/api/logto/user');
+  const { data: dataWithUserInfo } = useSWR<LogtoContext>('/api/logto/user-info');
```
Comment (charIeszhao) [inline packages/next-sample/pages/index.tsx:8]: Seems we also need to update these two routes

File: packages/node/src/index.ts [hunk 4]
```diff
@@ -79,12 +82,14 @@ export default class LogtoClient extends BaseClient {
...
-    } catch {
-      return {
-        isAuthenticated: false,
-      };
-    }
+    } catch {}
+
+    return {
+      isAuthenticated: false,
+    };
```
Comment (charIeszhao) [inline packages/node/src/index.ts:92]: Use `finally` instead?

**Review by charIeszhao — APPROVED** [review]

File: packages/next-sample/pages/index.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/next-sample/pages/index.tsx]: `/api/logto/user` is the SDK's built-in route (also recommended), so I don't change the name, instead, I add 2 comments on them.

**Pull request merged**